auth/utils.go

```diff
@@ -22,6 +22,13 @@ type PrivateKey struct {
 	Bytes []byte
 }
 
+func NewPrivateKeyFromED25519(pk ed25519.PrivateKey) *PrivateKey {
```

Comment:
nit: should be in ed25519.go since it's specific to ed25519 - although I think we can get rid of this completely (per your comment here https://github.com/ava-labs/hypersdk/pull/1798/files#r1848745797)